### DISCUSSION TITLE
Make factor.ts extend EventTarget so that render events may be captured and used to build an interactive editor

### DISCUSSION THREAD

**Discussion by dschere** [0xfe/vexflow#1525] [Ideas]
[factory.ts 4.1.0 modified to produce events](https://github.com/dschere/daves-composer/blob/5b696660d0c847f32d48b997a8bd68e2c554e75a/src/dep/vexflow-modifications/factory-4.1.0-modified.ts)

factory could generate events as they are being sent to the renderQ

`
private emitFactoryEvent(className: string, obj: any) {
     this.dispatchEvent(
        new CustomEvent('vexFlowFactoryEvent', { detail: {
          className,
          obj
        }})
     );
  }
...

this.emitFactoryEvent("StaveNote", note); 
this.renderQ.push(note); // ^^^ emit event prior rander
`

Note: if minification didn't mangle the class names I could use introspection and just have one call emitFactoryEvent inside the factory.draw function but sadly this isn't the case. this should be sufficient for users to trap events and create interactive music notation.

**Comment by rvilarl** [comment]
Do you know if this requires much processing power? @AaronDavidNewman would that be something useful for smoosic?

**Comment by ronyeh** [comment]
Items that can be placed on the staff have a CATEGORY property that might be helpful with your event dispatch.

`static get CATEGORY(): string`
